Automattic/jetpack
# Odyssey : Add a new upgrade nudge to old stats page

Addresses #28729.

## Proposed changes:


Adds an upgrade notice to the legacy stats page for anyone who is not yet running Odyssey Stats.

<img width="880" alt="SCR-20230221-ns5" src="https://user-images.githubusercontent.com/40267301/220299446-88acde13-e7aa-4893-b6ce-c586b47ae62e.png">

The surrounding HTML is generated server side so we look at the response and insert our upgrade nudge when needed. The CSS is dumped inline as well.

The CSS is mostly cribbed from the [Notice class](https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/components/components/notice/style.module.scss) in JS Packages → Components.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Check out this branch.
* Disable Odyssey Stats if enabled. — `wp-admin/admin.php?page=jetpack#/traffic?term=Jetpack%20Stats`
* Visit the legacy Stats page. — `wp-admin/admin.php?page=stats`
* Confirm the "Switch to New Stats" button is working. It should reload the page directly into Odyssey Stats.
* Disable Odyssey Stats again.
* Revisit the Stats page and confirm the legacy page is shown.
* Confirm the "Learn about Stats" link works.
* Confirm the notice can be dismissed via the "x" button.
